lib/pbench/server/api/resources/intake_base.py

```diff
@@ -335,95 +327,84 @@ def _put(self, args: ApiParams, request: Request, context: ApiContext) -> Respon
                     hash_md5 = hashlib.md5()
 
                     while True:
-                        chunk = request.stream.read(self.CHUNK_SIZE)
+                        chunk = access.stream.read(self.CHUNK_SIZE)
                         bytes_received += len(chunk)
-                        if len(chunk) == 0 or bytes_received > content_length:
+                        if len(chunk) == 0 or bytes_received > access.length:
                             break
                         ofp.write(chunk)
                         hash_md5.update(chunk)
             except OSError as exc:
                 if exc.errno == errno.ENOSPC:
-                    raise CleanupTime(
-                        HTTPStatus.INSUFFICIENT_STORAGE,
-                        f"Out of space on {tar_full_path.root}",
-                    )
+                    # NOTE: Werkzeug doesn't support status 509, so the abort
+                    # call in _dispatch will fail. Rather than figure out how
+                    # to fix that, just report as an internal error.
+                    raise APIInternalError(
+                        f"Out of space on {tar_full_path.root}"
+                    ) from exc
                 else:
-                    raise CleanupTime(
-                        HTTPStatus.INTERNAL_SERVER_ERROR,
-                        f"Unexpected error {exc.errno} encountered during file upload",
-                    )
-            except Exception:
-                raise CleanupTime(
-                    HTTPStatus.INTERNAL_SERVER_ERROR,
-                    "Unexpected error encountered during file upload",
-                )
+                    raise APIInternalError(
+                        f"Unexpected error {exc.errno} encountered during file upload"
+                    ) from exc
```

Comment:
Alternately,
```python
                # NOTE: Werkzeug doesn't support status 509, so the abort
                # call in _dispatch will fail. Rather than figure out how
                # to fix that, just report INSUFFICIENT_STORAGE as an
                # internal error.
                if exc.errno == errno.ENOSPC:
                    msg = f"Out of space on {tar_full_path.root}"
                else:
                    msg = f"Unexpected error {exc.errno} encountered during file upload"
                raise APIInternalError(msg) from exc

```

Reply:
I was bummed that Werkzeug won't let me report this the way I wanted; I think that I didn't want to make it look too easy. 😦